### PR TITLE
👴Clickable ancestors

### LINKED ISSUE #129: 👴Clickable ancestors
- [x] Crawl to the top clickable ancestors (short-circuit at data attribute) and use that as the target 
- [x] Test w/ 🗞️Data attribute example for NY Times -- QAW_DATA_ATTRIBUTE: data-testid

### PR DESCRIPTION
**Pull request by jperl**
Closes #129

Find highest clickable ancestor (stopping at closest w/ data attribute) and use that as the click target.

- [x] explain why we want to do this in a comment (svgs, icons, etc)
- [x] clean up API -- getComputedStyle is used in multiple places/should isClickable go in element?/etc...
- [x] test against many sites
- [x] add unit tests for findClickableAncestor

### CODE REVIEW

**Comment by jperl** [comment]
Relates to https://github.com/qawolf/examples/pull/1

File: packages/web/src/find.ts [hunk 1]
```diff
@@ -78,8 +115,9 @@ export const findElement = async ({
   if (target.xpath === "/html" || target.xpath === "/html/body")
     return findElementByXpath(target.xpath);
 
+  console.log("waiting for top strong match", target);
```
Comment (jperl) [inline packages/web/src/find.ts]: moved this outside of the waitFor so we don't have a bunch of repetitive logs

File: packages/web/src/element.ts [hunk 2]
```diff
@@ -163,3 +163,27 @@ export const getDescriptor = (
...
+};
+
+export const isVisible = (
+  element: HTMLElement,
+  computedStyle?: CSSStyleDeclaration
+): boolean => {
+  if (element.offsetWidth <= 0 || element.offsetHeight <= 0) return false;
+
+  // make sure the element is large enough to interact with
+  const rect = element.getBoundingClientRect();
```
Comment (flaurida) [inline packages/web/src/element.ts]: don't use this method it is very slow and non performant

File: packages/web/src/element.ts [hunk 3]
```diff
@@ -163,3 +163,27 @@ export const getDescriptor = (
...
+  // assume it is clickable if the cursor is a pointer
+  const clickable = computedStyle.cursor === "pointer";
+  return clickable && isVisible(element, computedStyle);
+};
+
+export const isVisible = (
+  element: HTMLElement,
+  computedStyle?: CSSStyleDeclaration
+): boolean => {
+  if (element.offsetWidth <= 0 || element.offsetHeight <= 0) return false;
```
Comment (flaurida) [inline packages/web/src/element.ts]: make sure using right `width` method: https://developer.mozilla.org/en-US/docs/Web/API/CSS_Object_Model/Determining_the_dimensions_of_elements

Comment (jperl) [reply packages/web/src/element.ts]: I didn't write that code :0. But it seems right to me.

**Pull request merged**